### DISCUSSION TITLE
moving a container while rofi is active switches workspaces

### DISCUSSION THREAD

**Discussion by berezowski** [davatorium/rofi#2045] [(Possible) Issues] [answered]
Hi,

Rofi Version: 1.7.5+wayland3

i wrote this workspace switcher that behaves strangely when it's being run by rofi and i suspect it has something to do with the return Signal magic that rofi does when selecting an entry (selected value + 10). 

I'm asking here because i have no clue how to debug this.
It works as intended when i run it from any shell.
It works as intended when executed by sway itself via a keybinding.

so here is the Problem:
sway's ipc interface is called with the correct command, equivalent to
swaymsg move container to workspace "Foo"
but additionally to moving the container the new workspace is selected, as well as sometimes selecting all windows on the original workspace.

I appreciate any serious input or guesses. Please have a look yourself:

https://github.com/user-attachments/assets/dba34b1f-4a12-43f9-8684-30005ea7c1ce

Thanks!
Hubert

**Comment by berezowski** [comment]
with rofi running i ran the sway keybinding i used to test the issue 
```bindsym Mod1+Ctrl+0 exec /usr/bin/swaymsg_workspace rofi_move_window hhh```
the new workspace was selected immediately. (Instead of just moving the window to the workspace).

Then i tried the 'standard' move keybinding with rofi visible.
```move container to workspace number 1```
Same, same. The Focus switched to workspace number 1.

So it turns out while rofi is visible moving containers behaves differently as when it is not:
- without rofi: it just moves the container
- with rofi visible: moving also selects the container

**Comment by DaveDavenport** [comment, marked as answer]
I recommend asking it  here: https://github.com/lbonn/rofi/  as that is the version you reportedly run.
This is very wayland specific, and this version has no wayland support.

**Reply by berezowski** [reply]
Oh, i'm sorry, i was not aware of that. Thanks
